src/rust/ide/src/controller/searcher/action/hardcoded.rs

```diff
@@ -0,0 +1,195 @@
+//! A module containing the hard-coded definitions displayed in Searcher. The main function to use
+//! is [`add_hardcoded_entries_to_list`] which adds the entries to given [`ListBuilder`].
+
+use crate::prelude::*;
+
+use crate::double_representation::module;
+use crate::double_representation::tp;
+use crate::controller::searcher::action;
+use crate::controller::searcher::action::ListBuilder;
+use crate::model::module::MethodId;
+
+
+
+// ===================
+// === Definitions ===
+// ===================
+
+// === RootCategory ===
+
+/// The hardcoded root category.
+#[allow(missing_docs)]
+#[derive(Clone,Debug)]
+pub struct RootCategory {
+    pub name       : &'static str,
+    pub categories : Vec<Category>,
+}
+
+
+// === Category ===
+
+/// The hardcoded second-tier category.
+#[allow(missing_docs)]
+#[derive(Clone,Debug)]
+pub struct Category {
```

Comment:
Same as above. Or perhaps a single note comment should be referred to from both structs.